hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoContextUtils.java

```diff
@@ -45,27 +46,25 @@ private CryptoContextUtils() {
 		return allowanceMap;
 	}
 
-	public static Map<ExtantCryptoContext.AllowanceMapKey, Long> convertToTokenMapFromGranted(
+	public static Map<AllowanceId, Long> convertToTokenMapFromGranted(
 			final List<GrantedTokenAllowance> allowances) {
-		Map<ExtantCryptoContext.AllowanceMapKey, Long> allowanceMap = new HashMap<>();
+		Map<AllowanceId, Long> allowanceMap = new HashMap<>();
 		for (var a : allowances) {
-			allowanceMap.put(new ExtantCryptoContext.AllowanceMapKey(a.getTokenId().getTokenNum(),
+			allowanceMap.put(new AllowanceId(a.getTokenId().getTokenNum(),
 					a.getSpender().getAccountNum()), a.getAmount());
 		}
 		return allowanceMap;
 	}
 
-	public static Map<ExtantCryptoContext.AllowanceMapKey, ExtantCryptoContext.AllowanceMapValue> convertToNftMapFromGranted(
+	public static Set<AllowanceId> convertToNftMapFromGranted(
 			final List<GrantedNftAllowance> allowances) {
-		Map<ExtantCryptoContext.AllowanceMapKey, ExtantCryptoContext.AllowanceMapValue> allowanceMap =
-				new HashMap<>();
+		Set<AllowanceId> approveForAllAllowances =
+				new TreeSet<>();
```

Comment:
do we need this to be a`TreeSet` or `HashSet` should be okay?

Reply:
a sorted set will help us in maintaining the order on all nodes when serializing and deserializing.